cfme/tests/infrastructure/test_infra_tag_filters_combination.py

```diff
@@ -0,0 +1,69 @@
+import pytest
+
+from cfme import test_requirements
+from cfme.infrastructure.cluster import Cluster
+from cfme.infrastructure.provider import InfraProvider
+from cfme.infrastructure.virtual_machines import Vm, Template
+from fixtures.provider import setup_one_or_skip
+from utils.providers import ProviderFilter
+from utils.update import update
+
+
+pytestmark = [test_requirements.tag, pytest.mark.tier(2)]
+
+test_items = [
+    ('clusters', Cluster),
+    ('vms', Vm),
+    ('templates', Template)
+]
+
+
+@pytest.fixture(scope='module')
+def a_provider(request):
+    prov_filter = ProviderFilter(classes=[InfraProvider],
+                                 required_fields=['datacenters', 'clusters'])
+    return setup_one_or_skip(request, filters=[prov_filter])
+
+
+@pytest.fixture(params=test_items, ids=[str(test_id[0]) for test_id in test_items],
+                scope='function')
+def testing_vis_object(request, a_provider, appliance):
+    """ Fixture creates class object for tag visibility test
+    Returns: class object of certain type
+    """
+    collection_name, param_class = request.param
+    test_items = getattr(appliance.rest_api.collections, collection_name)
+
+    if not test_items:
+        pytest.skip('No content found for test!')
+
+    if collection_name == 'templates':
+        item_type = a_provider.data['provisioning']['catalog_item_type'].lower()
+        for test_item_value in test_items:
```

Comment:
OPTIONAL: I would use ``resource`` or ``entity`` here since these are the terms that we use for items in collections wrt REST API.